### PR TITLE
Add log_format configuration (text|json) 

### LINKED ISSUE #3313: Feature request: eventhandler logs as json
We are currently successfully using the following integration:
```
    integrations:
      eventhandler:
        cache_path: "/etc/eventhandler/eventhandler.cache"
        logs_instance: "default"
```
Giving us the k8s events as logs in the application namespaces.
So far so good, however, it kind of interferes with our application logs, in the way that all our applications are standardised on `json` logs.

So suddenly, when querying logs with something such as `{namespace="foo"} | json` gives a json parse error because these event logs get in the way. 

Would it be possible for the eventhandler integration to output its logs in json format ? Maybe make the output format configurable?

### PR DESCRIPTION
**Pull request by sadovnikov**
#### PR Description

The `log_format: <string> | default = "logfmt"` configuration option became available for `eventhandler` and the `loki.source.kubernetes_events` Flow component

#### Which issue(s) this PR fixes

Resolves #3313

#### Notes to the Reviewer

> For whoever works on this: the change should be mirrored to the loki.source.kubernetes_events Flow component.

The issue includes the comment above. Not sure where to find the code for this

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Review by clayton-cornell — COMMENTED** [review]
Doc input is small. Is this enough to cover the change? Or do we need more explanation?

**Comment by sadovnikov** [comment]
I changed the default value of `log_format` to `logfmt` and slightly extended its description.

Both `json` and `logfmt` parsers are tested
```
{job="integrations/kubernetes/eventhandler"} | json | type = "Warning" | count > 100
```

image

**Comment by marctc** [comment]
> > For whoever works on this: the change should be mirrored to the loki.source.kubernetes_events Flow component.
> 
> The issue includes the comment above. Not sure where to find the code for this

The relevant code to change this is in `component/loki/source/kubernetes_events/kubernetes_events.go` and `component/loki/source/kubernetes_events/event_controller.go`. It would be great if have feature parity between the integration and the flow component as we are slowly recommending users to move to Flow.

File: docs/sources/flow/reference/components/loki.source.kubernetes_events.md [hunk 1]
```diff
@@ -26,11 +26,12 @@ log entries to the list of receivers passed in `forward_to`.
...
 `loki.source.kubernetes_events` supports the following arguments:
 
-Name | Type | Description | Default | Required
----- | ---- | ----------- | ------- | --------
+Name | Type | Description                                      | Default | Required
+---- | ---- |--------------------------------------------------| ------- | --------
 `job_name` | `string` | Value to use for `job` label for generated logs. | `"loki.source.kubernetes_events"` | no
-`namespaces` | `list(string)` | Namespaces to watch for Events in. | `[]` | no
-`forward_to` | `list(LogsReceiver)` | List of receivers to send log entries to. | | yes
+`log_format` | `string` | Log format, either "logfmt" or "json". | `"logfmt"` | no
```
Comment (marctc) [inline docs/sources/flow/reference/components/loki.source.kubernetes_events.md:32]: I'd personally leave minimal description (like "Format of the log") and then below the table explain a bit more in detail the function of this field and the possible values that it takes. You can reuse the text from `eventhandler-config.md`

File: component/loki/source/kubernetes_events/kubernetes_events.go [hunk 2]
```diff
@@ -43,14 +43,16 @@ type Arguments struct {
 
 	JobName    string   `river:"job_name,attr,optional"`
 	Namespaces []string `river:"namespaces,attr,optional"`
+	LogFormat  string   `river:"log_format,attr,optional"`
```
Comment (marctc) [inline component/loki/source/kubernetes_events/kubernetes_events.go:46]: I think it's worth to check in `Validate` that `LogFormat` introduced is either `logfmt` or `json`

**Review by marctc — APPROVED** [review]
Thanks a lot for contributing :pray:!! After @clayton-cornell review, I can merge it

**Review by clayton-cornell — APPROVED** [review]
Looks good/consistent with other ref topics :-)

@marctc, over to you to merge

**Pull request merged**